pandas/io/formats/style.py

```diff
@@ -722,15 +715,10 @@ def set_td_classes(self, classes: DataFrame) -> Styler:
         """
         classes = classes.reindex_like(self.data)
 
-        mask = (classes.isna()) | (classes.eq(""))
-        self.cell_context["data"] = {
-            r: {
-                c: [str(classes.iloc[r, c])]
-                for c, cn in enumerate(classes.columns)
-                if not mask.iloc[r, c]
-            }
-            for r, rn in enumerate(classes.index)
-        }
+        for r, row_tup in enumerate(classes.itertuples()):
```

Comment:
this will probably not be as performant, but may not matter. can you see.

Reply:
I tested and it was much better. I think because it eliminates multiple `.iloc` (line 730) lookups and it prevents doing this `cell_context.get("data", {}).get(r, {}).get(c, [])` (line 510) for every cell.

![Screen Shot 2021-03-20 at 07 32 02](https://user-images.githubusercontent.com/24256554/111861493-f4a76e80-894e-11eb-975c-e786b8220648.png)

Reply:
actually, might be worth having benchmarks for this since it is an alternative way of adding styles to cells.

```
       before           after         ratio
     [38640d11]       [c425ddda]
     <master>         <cell_context_default_dict>
-      11.5±0.1ms      6.60±0.08ms     0.57  io.style.RenderApply.time_classes_render(12, 12)
-      31.7±0.6ms       16.6±0.1ms     0.52  io.style.RenderApply.time_classes_render(36, 12)
-      79.1±0.9ms         30.6±3ms     0.39  io.style.RenderApply.time_classes_render(12, 120)
-         228±7ms       75.2±0.9ms     0.33  io.style.RenderApply.time_classes_render(36, 120)
-         155±2ms       51.3±0.6ms     0.33  io.style.RenderApply.time_classes_render(24, 120)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```

Reply:
great